x86_64 +softfloat crashes LLVM

## First bad commit
9bbd3ab

## Steps to reproduce
Attempt to build [Zigvale Barebones](https://github.com/ominitay/zigvale-barebones) or [Florence OS](https://github.com/FlorenceOS/Florence/).

## Expected Behaviour
The projects build successfully. 

## Actual Behaviour
The compiler crashes in release mode, or segfaults in debug mode. 

cc: @N00byEdge 
cc: @Vexu

I have been thinking about it and the example that you mentioned shouldn't crash in LLVM (I don't know if I am wrong, but that's my guess). If you remove +soft_float, it doesn't get to LLVM and it gives a compilation error: 
```
./zig-linux-x86_64-0.10.0-dev.871+bb05a8a08/lib/std/start.zig:546:45: error: root source file has no member called 'main'
    switch (@typeInfo(@typeInfo(@TypeOf(root.main)).Fn.return_type.?)) {
```
Shouldn't it give it with +soft_float too? My guess this is another different kind of error.

In fact, if you compile this with 0.9.1, it also seg faults, which indicates that something was wrong there before the bug of this issue was introduced (My guess again is that Zig should give a compiler error before getting to LLVM, but I might be wrong).

Thanks everybody for contributing and I hope my comment helps you too!

--- a/src/codegen/llvm.zig
+++ b/src/codegen/llvm.zig
@@ -9514,7 +9514,7 @@ fn isByRef(ty: Type) bool {
 /// and false if we expect LLVM to crash if it counters an x86_fp80 type.
 fn backendSupportsF80(target: std.Target) bool {
     return switch (target.cpu.arch) {
-        .x86_64, .i386 => true,
+        .x86_64, .i386 => !std.Target.x86.featureSetHas(target.cpu.features, .soft_float),
         else => false,
     };
 }

